Src/VimCore/CommandUtil.fs

```diff
@@ -1836,31 +1842,36 @@ type internal CommandUtil
 
                 // Now run the put after command
                 let register = x.GetRegister (Some RegisterName.Unnamed)
-                x.PutAfterCaretCore register.RegisterValue 1 false
+                x.EditWithUndoTransaction "Put after mouse" (fun () ->
+                    x.PutAfterCaretCore register.RegisterValue 1 false)

```

Comment:
Why did we need to switch to explicit transactions in this set of APIs?

Reply:
This is because when you put from virtual space, we have to fill in virtual spaces first, otherwise the put gets jammed onto the end of the line it was on. So we we want both the real spaces added (if any) and the put text to be part of the same transaction.